py/picca/delta_extraction/least_squares/least_squares_var_stats.py

```diff
@@ -169,21 +174,14 @@ def initialize_delta_arrays(self, forests):
             bins = var_pipe_bins + NUM_VAR_BINS * log_lambda_bins
 
             # compute deltas
-            delta = (forest.flux / forest.continuum - 1)
-            delta = delta[w]
+            delta = forest.flux[w] / forest.continuum[w] - 1
 
             # add contributions to delta statistics
-            rebin = np.bincount(bins, weights=delta)
-            mean_delta[:len(rebin)] += rebin
-
-            rebin = np.bincount(bins, weights=delta**2)
-            var_delta[:len(rebin)] += rebin
-
-            rebin = np.bincount(bins, weights=delta**4)
-            var2_delta[:len(rebin)] += rebin
+            mean_delta += np.bincount(bins, weights=delta, minlength=mean_delta.size)
```

Comment:
Not sure these changes will not crash in some cases. Why can we now drop the `[:len(rebin)]`?

Reply:
It shouldn't crash. minlength argument makes sure the array `np.bincount` returns is the same size as `mean_delta` (since there are no overflows). Without this argument `np.bincount` removes the array at the right end, so you needed `[:len(rebin)]`.

Reply:
isn't minlength just ensuring it's at least that size? But agreed, it might be that now that we're not dropping pixels, but keeping them some of those [:...] statements can be removed. I'd need to think about this in more depth, to fully grasp where this is now possible and where not.